python/pyiceberg/catalog/rest.py

```diff
@@ -0,0 +1,365 @@
+            "destination": self._split_namespace_and_name(to_identifier),
+        }
+        response = requests.post(self.url(Endpoints.rename_table), json=payload, headers=self.headers)
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchTableError, 409: TableAlreadyExistsError})
+
+    def create_namespace(self, namespace: Union[str, Identifier], properties: Optional[Properties] = None) -> None:
+        payload = {"namespace": self.identifier_to_tuple(namespace), "properties": properties}
+        response = requests.post(self.url(Endpoints.create_namespace), json=payload, headers=self.headers)
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {409: AlreadyExistsError})
+
+    def drop_namespace(self, namespace: Union[str, Identifier]) -> None:
+        namespace = NAMESPACE_SEPARATOR.join(self.identifier_to_tuple(namespace))
+        response = requests.delete(self.url(Endpoints.drop_namespace, namespace=namespace), headers=self.headers)
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchNamespaceError})
+
+    def list_namespaces(self) -> List[Identifier]:
+        response = requests.get(self.url(Endpoints.list_namespaces), headers=self.headers)
+        response.raise_for_status()
+        namespaces = ListNamespaceResponse(**response.json())
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {})
+        return namespaces.namespaces
+
+    def load_namespace_properties(self, namespace: Union[str, Identifier]) -> Properties:
+        namespace = NAMESPACE_SEPARATOR.join(self.identifier_to_tuple(namespace))
+        response = requests.get(self.url(Endpoints.load_namespace_metadata, namespace=namespace), headers=self.headers)
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchNamespaceError})
+
+        return NamespaceResponse(**response.json()).properties
+
+    def update_namespace_properties(
+        self, namespace: Union[str, Identifier], removals: Optional[Set[str]] = None, updates: Optional[Properties] = None
+    ) -> UpdateNamespacePropertiesResponse:
+        namespace = NAMESPACE_SEPARATOR.join(self.identifier_to_tuple(namespace))
+        payload = {"removals": list(removals or []), "updates": updates}
+        response = requests.post(self.url(Endpoints.update_properties, namespace=namespace), json=payload, headers=self.headers)
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchNamespaceError})
+        return UpdateNamespacePropertiesResponse(**response.json())
```

Comment:
Why update the method's type to be able to return an internal response class?

Reply:
I've created a new class next to the interface. I agree that returning to an internal response class isn't the best idea, and meant to clean that up. The REST API returns which fields are being removed, updated, or missing. That's something that we definitely want to expose in a CLI or in orchestrators.

Reply:
Does this need to return anything at all? I'm okay either way, I think.

Reply:
Yes, I would like that, for the abovementioned reason 👍🏻